packages/@lwc/integration-karma/test/template/attribute-class/object-values.spec.js

```diff
@@ -0,0 +1,129 @@
+import { createElement } from 'lwc';
+
+import Dynamic from 'x/dynamic';
+import Reactive from 'x/reactive';
+
+function createDynamicClass(value) {
+    const elm = createElement('x-dynamic', { is: Dynamic });
+    elm.dynamicClass = value;
+    document.body.appendChild(elm);
+
+    return {
+        host: elm,
+        target: elm.shadowRoot.querySelector('div'),
+    };
+}
+
+function testClassNameValue(name, value, expected) {
+    it(name, () => {
+        const { target } = createDynamicClass(value);
+        expect(target.className).toBe(expected);
+    });
+}
+
+function testReactiveClassNameValue(name, setupFn, updateFn, expected) {
+    it(name, () => {
+        const elm = createElement('x-reactive', { is: Reactive });
+        elm.updateDynamicClass(setupFn);
+        document.body.appendChild(elm);
+
+        const target = elm.shadowRoot.querySelector('div');
+
+        elm.updateDynamicClass(updateFn);
+        return Promise.resolve().then(() => {
+            expect(target.className).toBe(expected);
+        });
+    });
+}
+
+// Object class binding is enabled starting from API v61.
+const isObjectBindingEnabled = process.env.API_VERSION >= 61;
+
+describe('type coercion', () => {
+    testClassNameValue('object', {}, isObjectBindingEnabled ? '' : '[object Object]');
+    testClassNameValue('true', true, isObjectBindingEnabled ? '' : 'true');
+    testClassNameValue('false', false, isObjectBindingEnabled ? '' : 'false');
+    testClassNameValue('null', null, isObjectBindingEnabled ? '' : '');
+    testClassNameValue('undefined', undefined, isObjectBindingEnabled ? '' : '');
+    testClassNameValue('number', 1, isObjectBindingEnabled ? '' : '1');
```

Comment:
I'd like to see `0` and `NaN` here as well for completeness, since they are falsy.